### PR TITLE
feat: emit slashing events

### LINKED ISSUE #6209: Add proposer_slashing and attester_slashing to SSE
### Problem description

Reference: https://github.com/ethereum/beacon-APIs/pull/376

### Solution description

Events API should now include proposer and attester slashing information. 

### Additional context

This will allow our node operators to easily detect slashing events for use with tooling like https://github.com/attestantio/esd.

### PR DESCRIPTION
**Pull request by jeluard**
**Motivation**

Add `proposer_slashing` and `attester_slashing` to SSE 

**Description**

Make sure new events introduced by https://github.com/ethereum/beacon-APIs/pull/376 are now emitted and made available via SSE.

Closes #6209

### CODE REVIEW

**Review by wemeetagain — COMMENTED** [review]
this lgtm, ready for review?

File: packages/api/src/beacon/routes/events.ts [hunk 1]
```diff
@@ -35,6 +35,10 @@ export enum EventType {
   attestation = "attestation",
   /** The node has received a valid voluntary exit (from P2P or API) */
   voluntaryExit = "voluntary_exit",
+  /** The node has received a ProposerSlashing (from P2P or API) that passes validation rules of the `proposer_slashing` topic */
+  proposerSlashing = "proposer_slashing",
+  /** The node has received an AttesterSlashing (from P2P or API) that passes validation rules of the `attester_slashing` topic */
+  attesterSlashing = "attester_slashing",
```
Comment (nflaig) [inline packages/api/src/beacon/routes/events.ts]: I'd suggest to align this with existing documentation

```suggestion
  /** The node has received a valid proposer slashing (from P2P or API) */
  proposerSlashing = "proposer_slashing",
  /** The node has received a valid attester slashing (from P2P or API) */
  attesterSlashing = "attester_slashing",
```

File: packages/beacon-node/test/e2e/network/gossipsub.test.ts [hunk 2]
```diff
@@ -138,6 +138,70 @@ function runTests({useWorker}: {useWorker: boolean}): void {
     );
   });
 
+  it("Publish and receive an attesterSlashing", async function () {
```
Comment (nflaig) [inline packages/beacon-node/test/e2e/network/gossipsub.test.ts:141]: The amount of copy pasted code is quite high in this file, maybe something that could be improved in this PR (or later one) to make it easier to add new events in the future

Comment (jeluard) [reply packages/beacon-node/test/e2e/network/gossipsub.test.ts:141]: I would suggest to do that in a different PR for extra clarity.

**Review by nflaig — APPROVED** [review]

**Review by nflaig — APPROVED** [review]

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.16.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.16.0) :tada:

**Pull request merged**